File: 864-image-overlap/image-overlap.js
// declare two variables arr1 and arr2 that will hold on to values from our iteration
// get the dimensions of img1 and img2
// iterate through both images
// if the current cell is a 1, push the dimensions into the respective arrays
// declare a variable lookup that will store the results of the translations
// iterate through arr1 and arr2
// store the diff between dimensions in lookup
// update the maximum overlap so far
// return the overlap at the end of the iterations

var largestOverlap = function(img1, img2) {
    let arr1 = []
    let arr2 = []
    let rows = img1.length
    let columns = img1[0].length
    let lookup = {}
    let overlap = 0

    for (let i = 0; i < rows; i++) {
        for (let j = 0; j < columns; j++) {
            if (img1[i][j] == 1) arr1.push([i, j])
            if (img2[i][j] == 1) arr2.push([i, j])
        }
    }    

    for (let [x, y] of arr1) {
        for (let [i, j] of arr2) {
            translation = [i - x, j - y]
            if (!lookup[translation]) lookup[translation] = 0
            lookup[translation] ++
            overlap = Math.max(overlap, lookup[translation])
        }
    }
    return overlap
};

File: 864-image-overlap/image-overlap.py
# declare a lookup variable that will store the freq of cordinates
# declare a variable max that will hold the largest possible overlap
# iterate over both images and store their cordinates in tuple pairs
# iterate over both list of tuples
# if the difference is present in the lookup object increment it and update the max
# else add it to the lookup table

class Solution:
    def largestOverlap(self, img1: List[List[int]], img2: List[List[int]]) -> int:
        lookup = {}
        largest = 0
        rows = len(img1)
        columns = len(img1[0])
        list1 = []
        list2 = []

        for i in range(rows):
            for j in range(columns):
                if img1[i][j] == 1:
                    list1.append((i, j))
                if img2[i][j] == 1:
                    list2.append((i, j))

        for i, j in list1:
            for x, y in list2:
                translation = (i - x, j - y)
                if not translation in lookup:
                    lookup[translation] = 0
                lookup[translation] += 1
                largest = max(largest, lookup[translation]) 

        return largest


                
        